geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/QueryPlanner.scala

```diff
@@ -179,32 +178,32 @@ object QueryPlanner extends LazyLogging {
   def setQueryTransforms(query: Query, sft: SimpleFeatureType): Unit = {
     // even if a transform is not specified, some queries only use a subset of attributes
     // specify them here so that it's easier to pick the best column group later
-    def transformsFromQueryType: Seq[String] = {
+    def fromQueryType: Option[Seq[String]] = {
       val hints = query.getHints
       if (hints.isBinQuery) {
-        BinAggregatingScan.propertyNames(hints, sft)
+       Some(BinAggregatingScan.propertyNames(hints, sft))
       } else if (hints.isDensityQuery) {
-        DensityScan.propertyNames(hints, sft)
+        Some(DensityScan.propertyNames(hints, sft))
       } else if (hints.isStatsQuery) {
-        val props = StatParser.propertyNames(sft, hints.getStatsQuery)
-        if (props.nonEmpty) { props } else {
-          // some stats don't require explicit props (e.g. count), so just take a field that is likely
-          // to be available anyway
-          val prop = Option(sft.getGeomField)
-              .orElse(sft.getDtgField)
-              .orElse(FilterHelper.propertyNames(query.getFilter, sft).headOption)
-              .getOrElse(sft.getDescriptor(0).getLocalName)
-          Seq(prop)
-        }
+        Some(StatParser.propertyNames(sft, hints.getStatsQuery))
       } else {
-        Seq.empty
+        None
       }
     }
 
-    Option(query.getPropertyNames).map(_.toSeq)
-        .filter(_ != sft.getAttributeDescriptors.asScala.map(_.getLocalName))
-        .orElse(Some(transformsFromQueryType).filter(_.nonEmpty))
-        .foreach { props =>
+    // since we do sorting on the client, just add any sort-by attributes to the transform
+    // TODO GEOMESA-2655 we should sort and then transform back to the requested props, but it's complicated...
```

Comment:
Given that returning more columns/attributes than requested may not be expected, should we document this somehow?

Reply:
it's documented in the ticket :D
maybe we should have a known bugs section? but seems somewhat redundant with jira

Reply:
maybe we could have a link to JIRA 'affects version' searches for each version, like:
```
Known Bugs: 2.0.0, 2.0.1, 2.0.2, ...
```

Reply:
*shrugs*  Not sure what's a best practice here.